### ISSUE TITLE
Softmax of scores across channels / Baseline image

### ISSUE THREAD

**Issue by SirRob1997** [haofanwang/Score-CAM#6]
I have two questions and I couldn't find lines implementing the following functionality in [`scorecam.py`](https://github.com/haofanwang/Score-CAM/blob/master/cam/scorecam.py):

- In Algorithm 1 in the paper you compute the score using a baseline image X_b. This is not done [here](https://github.com/haofanwang/Score-CAM/blob/master/cam/scorecam.py#L64) and instead, we only have the first part of the equation. 

- In Algorithm 1 in the paper you apply softmax channel-wise for the importance scores. This is not done [here](https://github.com/haofanwang/Score-CAM/blob/master/cam/scorecam.py#L66) and instead, we directly multiply with the score.

Am I missing something?

**Comment by haofanwang** [comment]
Hi, @SirRob1997

Thanks for interests in our work.

1. We use a black image as baseline, so we don't include it in code.
2. The score is obtained after Softmax function, so it is a value within 0 to 1. We don't directly use the logit.

**Comment by SirRob1997** [comment]
For 1: Even for a black image, the score of a  class will not be zero, due to the learned biases. Given that it is a constant though, it doesn't matter too much, you're right.

For 2: The softmax is applied to the output of the model hence we softmax across the classes, obviously getting values [0,1]. However, this does not ensure that the scores across channels add up to 1 as you wrote in Algorithm 1.

**Comment by haofanwang** [comment]
Hi, @SirRob1997 , sorry for my late reply (I have been working on my CVPR submission).

You are right, this should be a typo and I think it does not matter. We just want to limit that every weight is within the same range.
